tests/integration/cocoon/test_seed_sample_data.py

```diff
@@ -0,0 +1,276 @@
+            "The provided file_type of bad_file_type has no associated mapping",
+        )
+
+    def test_instruments_from_response(self):
+        # In this test we take a random sample from the dataframe.
+        # We need to make a seperate get_instrument request for each item in the sample, so the test might become
+        # inefficient if we take the entire dataframe.
+        # A sample of approximately 10 should prove the function works as expected.
+
+        instruments_api = self.api_factory.build(lusid.api.InstrumentsApi)
+
+        random_10_ids = set(
+            random.choices(self.sample_data["instrument_id"].to_list(), k=10)
+        )
+        response_from_random_10 = set(
+            [
+                response.identifiers["ClientInternal"]
+                for response in [
+                    instruments_api.get_instrument(
+                        identifier_type="ClientInternal", identifier=id
+                    )
+                    for id in random_10_ids
+                ]
+            ]
+        )
+
+        self.assertEqual(random_10_ids, response_from_random_10)
+
+    def test_sub_holding_keys(self):
+        holdings_response = self.api_factory.build(
+            lusid.api.TransactionPortfoliosApi
+        ).get_holdings(
+            scope=self.scope, code=self.sample_data["portfolio_code"].to_list()[0],
+        )
+
+        list_of_prop_values = [
+            holding.sub_holding_keys[f"Transaction/{self.scope}/strategy"].value
+            for holding in holdings_response.values
+        ]
+        unique_strategy_labels = set([prop.label_value for prop in list_of_prop_values])
+
+        self.assertEqual(
+            unique_strategy_labels, set(self.sample_data["strategy"].to_list())
+        )
+
+
+class CocoonTestSeedDataNoOverrideCSV(unittest.TestCase, CocoonSeedDataTestsBase):
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        cls.scope = create_scope_id().replace("-", "_")
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+
+        cls.sample_data = pd.read_csv(
+            Path(__file__).parent.joinpath("data/seed_sample_data/sample_data.csv")
+        )
+
+        cls.domain_list = ["portfolios", "instruments", "transactions"]
+
+        cls.seed_data = seed_data(
+            cls.api_factory,
+            cls.domain_list,
+            cls.scope,
+            Path(__file__).parent.joinpath("data/seed_sample_data/sample_data.csv"),
+            sub_holding_keys=[f"Transaction/{cls.scope}/strategy"],
+            file_type="csv",
+        )
+
+    def test_return_dict(self):
+
+        seed_data = self.seed_data
+
+        self.assertEqual(type(seed_data), dict)
+        self.assertEqual(set(seed_data.keys()), set(self.domain_list))
+
+
+class CocoonTestSeedDataWithOverrideCSV(unittest.TestCase, CocoonSeedDataTestsBase):
```

Comment:
rename to CocoonTestSeedDataWithMappingOverrideCSV

Reply:
Done.